packages/expo-crypto/ios/EXCrypto/EXCrypto.m

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <EXCrypto/EXCrypto.h>
+#import <CommonCrypto/CommonDigest.h>
+
+@implementation EXCrypto
+
+EX_EXPORT_MODULE(ExpoCrypto);
+
+EX_EXPORT_METHOD_AS(digestStringAsync,
+                    digestStringAsync:(NSString *)algorithm
+                    data:(NSString *)data
+                    options:(NSDictionary *)options
+                    resolver:(EXPromiseResolveBlock)resolve
+                    rejecter:(EXPromiseRejectBlock)reject)
+{
+  NSString *encoding = options[@"encoding"];
+
+  int hashingAlgorithm = [self.class SHAJSONToNative:algorithm];
```

Comment:
```suggestion
  int digestLength = [EXCrypto digestLengthForJSAlgorithm:algorithm];
```
- `for (int i = 0; i < hashingAlgorithm; i++)` is confusing -> rename to digestLength
- follow `[aaaForBBB:bbb]` Cocoa convention